### PR TITLE
New samples: Kueue cohort tutorial

### PR DESCRIPTION
**Pull request by rbarberop**
This PR contains the yaml examples for the new Kueue tutorial using cohorts for resource sharing.

Code is v1beta1

### CODE REVIEW

**Comment by rbarberop** [comment]
@alculquicondor FYI

**Comment by rbarberop** [comment]
@Shabirmean https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/pull/498 closed in favor of this one

File: batch/kueue-cohort/team-a-job.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+metadata:
+  namespace: team-a # Job under team-a namespace
+  generateName: sample-team-a-job-
+  labels:
+    kueue.x-k8s.io/queue-name: team-a-lq # Point to the LocalQueue
+spec:
+  ttlSecondsAfterFinished: 60 # Job will be deleted after 60 seconds
+  parallelism: 3 # This Job will have 3 replicas running at the same time
+  completions: 3 # This Job requires 3 completions
+  suspend: true # Set to true and allow Kueue to control the Job
```
Comment (alculquicondor) [inline batch/kueue-cohort/team-a-job.yaml:27]:
```suggestion
  suspend: true # Set to true and allow Kueue to control when the Job starts
```

File: batch/kueue-cohort/team-b-job.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+metadata:
+  namespace: team-b # Job under team-a namespace
+  generateName: sample-team-b-job-
+  labels:
+    kueue.scheduling.k8s.io/queue-name: team-b-lq # Point to the LocalQueue
+spec:
+  ttlSecondsAfterFinished: 60 # Job will be deleted after 60 seconds
+  parallelism: 3 # This Job will have 3 replicas running at the same time
+  completions: 3 # This Job requires 3 completions
+  suspend: true # Set to true and allow Kueue to control the Job
```
Comment (alculquicondor) [inline batch/kueue-cohort/team-b-job.yaml:27]: same comment

**Review by alculquicondor — COMMENTED** [review]
/lgtm

**Review by bourgeoisor — APPROVED** [review]
LGTM, let me know when this is good to merge.

File: batch/kueue-cohort/team-a-cq.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+# limitations under the License.
+
+# [START gke_batch_kueue_cohort_team_a_cq]
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: ClusterQueue
+metadata:
+  name: team-a-cq
+spec:
+  cohort: all # team-a-cq and team-b-cq share the same cohort
+  namespaceSelector: {}
```
Comment (alculquicondor) [inline batch/kueue-cohort/team-a-cq.yaml:22]: this namespaceSelector should match the team-a namespace.

This way, users from team-b can't directly submit jobs here (they can still borrow, through team-b-cq)

Comment (alculquicondor) [reply batch/kueue-cohort/team-a-cq.yaml:22]:
```yaml
namespaceSelector:
  matchLabels:
    kubernetes.io/metadata.name: team-a
```

File: batch/kueue-cohort/spot-cq.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+# limitations under the License.
+
+# [START gke_batch_kueue_cohort_spot_cq]
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: ClusterQueue
+metadata:
+  name: spot-cq
+spec:
+  cohort: all
+  namespaceSelector: {}
```
Comment (alculquicondor) [inline batch/kueue-cohort/spot-cq.yaml:22]: you could remove this line: Then noboby is able to send jobs directly to this clusterqueue, but clusterqueues in the cohort can borrow.

File: batch/kueue-cohort/flavors.yaml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  nodeLabels:
+    spot: "false" # This label was applied when the cluster was created
+---
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: ResourceFlavor
+metadata:
+  name: spot # This ResourceFlavor will be used as added resource for the CPU resource
+spec:
+  nodeLabels:  
+    spot: "true" # This label was applied when the nodepool was created
```
Comment (alculquicondor) [inline batch/kueue-cohort/flavors.yaml:30]:
```yaml
cloud.google.com/gke-provisioning: spot
```

File: batch/kueue-cohort/flavors.yaml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+# limitations under the License.
+
+# [START gke_batch_kueue_cohort_flavors]
+apiVersion: kueue.x-k8s.io/v1beta1
+kind: ResourceFlavor
+metadata:
+  name: standard # This ResourceFlavor will be used for the CPU resource
+spec:
+  nodeLabels:
+    spot: "false" # This label was applied when the cluster was created
```
Comment (alculquicondor) [inline batch/kueue-cohort/flavors.yaml:22]:
```yaml
cloud.google.com/gke-provisioning: standard
```

File: batch/kueue-cohort/team-b-job.yaml [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+# limitations under the License.
+
+# [START gke_batch_kueue_cohort_team_b_job]
+apiVersion: batch/v1
+kind: Job
+metadata:
+  namespace: team-b # Job under team-a namespace
+  generateName: sample-team-b-job-
+  labels:
+    kueue.scheduling.k8s.io/queue-name: team-b-lq # Point to the LocalQueue
```
Comment (alculquicondor) [inline batch/kueue-cohort/team-b-job.yaml:22]:
```suggestion
    kueue.x-k8s.io/queue-name: team-b-lq # Point to the LocalQueue
```

File: batch/kueue-cohort/create_jobs.sh [hunk 8]
```diff
@@ -15,6 +15,10 @@
 #!/bin/bash
 while :
 do
+  kubectl create -f ${1}
+  kubectl create -f ${1}
+  kubectl create -f ${1}
+  kubectl create -f ${1}
```
Comment (alculquicondor) [inline batch/kueue-cohort/create_jobs.sh:21]: did you intend to have all these lines? You should be able to control the rate by changing the sleep time

**Review by alculquicondor — COMMENTED** [review]
LGTM

**Comment by rbarberop** [comment]
Blocked on https://github.com/kubernetes-sigs/kueue/issues/659

**Pull request merged**